js/utils.js

```diff
@@ -1025,6 +1025,178 @@ Fliplet.Registry.set('dynamicListUtils', (function () {
+      var bValue = b.data ? b.data[sortField] : b[sortField];
+      
+      switch (sortType) {
+        case 'string': 
+          /**
+           * By adding '{' to the start of the string we are pushing that string to the end the sort order,
+           * and when we are adding '}' we push it even further to the end
+          */
+          aValue = aValue ? aValue.toUpperCase() : '}';
+          bValue = bValue ? bValue.toUpperCase() : '}';
+
+          if (!startsWithAlphabet.test(bValue)) {
+            bValue = '{' + bValue;
+          }
+
+          if (!startsWithAlphabet.test(aValue)) {
+            aValue = '{' + aValue;
+          }
+          
+          if (aValue < bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+          if (aValue > bValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          return 0;
+        case 'number':
+          if (!parseInt(aValue, 10)) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          if (!parseInt(bValue, 10)) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (isSortAsc) {
+            return aValue - bValue;
+          }
+
+          return bValue - aValue;
+        case 'time':
+          if (!aValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          if (!bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (aValue < bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (aValue > bValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          return 0;
+        case 'date':
+          aValue = moment(aValue).valueOf();
+          bValue = moment(bValue).valueOf();
+
+          if (!aValue) {
+            return isSortAsc ? 1 : -1;
+          }
+
+          if (!bValue) {
+            return isSortAsc ? -1 : 1;
+          }
+
+          if (isSortAsc) {
+            return aValue - bValue;
+          }
+
+          return bValue - aValue;
+         default: 
+          if (isSortAsc) {
+            return aValue - bValue;
+          }
+
+          return bValue - aValue;
+      }
+    });
+  }
+
+  function sortRecordsByField(options) {
+    var sortedRecords = sortByField(options);
+
+    if (options.onlyRecords) {
+      return sortedRecords;
+    }
+
+    sortHTMLElements({
+      $layoutContainer: options.$container,
+      listContainer: options.listContainer,
+      listHolder: options.listHolder,
+      sortedRecords: sortedRecords
+    });
+  }
+
+  function sortHTMLElements(options) {
+    var sortedItemsList = [];
+    var $list = $(options.listContainer);
+    var $prevElement = $list.prev();
+    var $detachedList = $list.detach();
+    var $listItems = $detachedList.find('.small-card-list-item').detach();
+
+    $listItems.each(function() {
+      var $listItem = $(this);
+      var itemId = parseInt($listItem.data('entry-id'), 10);
+      var itemSortedIndex = options.sortedRecords.findIndex(function(record) {
```

Comment:
Not supported in IE.

```suggestion
      var itemSortedIndex = _.findIndex(options.sortedRecords, function(record) {
```

Reply:
This still needs action

Reply:
Fixed